### PR TITLE
Functests: Add a custome junit and a k8s reporter

### PR DESCRIPTION
**Pull request by liornoy**
This PR adds custom JUnit and k8s reporters to the latency and performance e2e test suites.
The reporters are created only if the `-junit` and/or `-report` flags are passed with the path
for the dir where to create the output. **The directory must be created already**

Since we're working on splitting the cnf-tests, we must keep producing the same reports for
failed test cases with a k8s reporter, and a custom junit report.

This PR modifies the test suites that are invoked by the cnf-tests and adds those reporters.

### Example of reports dir output:
image

#### With default junit-report:
```
<testcase name="[BeforeSuite]" classname="Performance Test" status="passed" time="0.002587849">
              <system-err>&gt; Enter [BeforeSuite] TOP-LEVEL - /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/test_suite_performance_test.go:24 @ 05/22/23 10:20:46.633&#xA;&lt; Exit [BeforeSuite] TOP-LEVEL - /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/test_suite_performance_test.go:24 @ 05/22/23 10:20:46.635 (3ms)&#xA;</system-err>
          </testcase>
<testcase name="[It] [performance]Hugepages [rfe_id:27369]when NUMA node specified [test_id:27752][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] should be allocated on the specifed NUMA node" classname="Performance Test" status="failed" time="0.006851844">
              <failure message="Expected&#xA;    &lt;[]v1.Node | len:0, cap:0&gt;: nil&#xA;not to be empty" type="failed">[FAILED] Expected&#xA;    &lt;[]v1.Node | len:0, cap:0&gt;: nil&#xA;not to be empty&#xA;In [BeforeEach] at: /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/hugepages.go:49 @ 05/22/23 10:20:46.642&#xA;</failure>
              <system-err>&gt; Enter [BeforeEach] [performance]Hugepages - /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/utils.go:24 @ 05/22/23 10:20:46.635&#xA;&lt; Exit [BeforeEach] [performance]Hugepages - /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/utils.go:24 @ 05/22/23 10:20:46.637 (2ms)&#xA;&gt; Enter [BeforeEach] [performance]Hugepages - /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/hugepages.go:39 @ 05/22/23 10:20:46.637&#xA;[FAILED] Expected&#xA;    &lt;[]v1.Node | len:0, cap:0&gt;: nil&#xA;not to be empty&#xA;In [BeforeEach] at: /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/hugepages.go:49 @ 05/22/23 10:20:46.642&#xA;&lt; Exit [BeforeEach] [performance]Hugepages - /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/hugepages.go:39 @ 05/22/23 10:20:46.642 (5ms)&#xA;</system-err>
          </testcase>
<testcase name="[It] [performance]Hugepages with multiple sizes [test_id:34080] should be supported and available for the container usage" classname="Performance Test" status="skipped" time="0">
              <skipped message="skipped - Spec skipped because an earlier spec in an ordered container failed"></skipped>
              <system-err>[SKIPPED] Spec skipped because an earlier spec in an ordered container failed&#xA;In [It] at: /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/hugepages.go:84 @ 05/22/23 10:20:46.642&#xA;</system-err>
          </testcase>
```
#### with the custom configurations, we bring here:
```
<testcase name="[performance]Hugepages [rfe_id:27369]when NUMA node specified [test_id:27752][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] should be allocated on the specifed NUMA node" classname="Performance Test" status="failed" time="0.002255927">
              <failure message="Expected&#xA;    &lt;[]v1.Node | len:0, cap:0&gt;: nil&#xA;not to be empty" type="failed">[FAILED] Expected&#xA;    &lt;[]v1.Node | len:0, cap:0&gt;: nil&#xA;not to be empty&#xA;In [BeforeEach] at: /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/hugepages.go:49 @ 05/22/23 10:11:31.026&#xA;</failure>
              <system-err>&gt; Enter [BeforeEach] [performance]Hugepages - /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/utils.go:24 @ 05/22/23 10:11:31.024&#xA;&lt; Exit [BeforeEach] [performance]Hugepages - /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/utils.go:24 @ 05/22/23 10:11:31.025 (1ms)&#xA;&gt; Enter [BeforeEach] [performance]Hugepages - /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/hugepages.go:39 @ 05/22/23 10:11:31.025&#xA;[FAILED] Expected&#xA;    &lt;[]v1.Node | len:0, cap:0&gt;: nil&#xA;not to be empty&#xA;In [BeforeEach] at: /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/hugepages.go:49 @ 05/22/23 10:11:31.026&#xA;&lt; Exit [BeforeEach] [performance]Hugepages - /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/hugepages.go:39 @ 05/22/23 10:11:31.026 (1ms)&#xA;&gt; Enter [ReportAfterEach] TOP-LEVEL - /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/test_suite_performance_test.go:82 @ 05/22/23 10:11:31.026&#xA;&lt; Exit [ReportAfterEach] TOP-LEVEL - /home/lnoy/go/src/github.com/liornoy/net/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/1_performance/test_suite_performance_test.go:82 @ 05/22/23 10:11:56.627 (25.601s)&#xA;</system-err>
          </testcase>
<testcase name="[performance]Hugepages with multiple sizes [test_id:34080] should be supported and available for the container usage" classname="Performance Test" status="skipped" time="0">
              <skipped message="skipped - Spec skipped because an earlier spec in an ordered container failed"></skipped>
          </testcase>
```

A few changes are visible here:
1. No entry for the BeforeSuite spec.
2. Test cases are titled without [It].
3. no `<system-err>` for skipped test cases.

### CODE REVIEW

**Comment by liornoy** [comment]
@ffromani notice I added a commit adding a k8s reporter. and holding this PR until we will merge a dependency conflict on the k8sreporter: https://github.com/openshift-kni/k8sreporter/pull/5

File: test/e2e/k8sreporter/reporter.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	performancev1alpha1 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v1alpha1"
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+
+	"github.com/openshift-kni/k8sreporter"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func New(kubeconfig, path, namespace string) *k8sreporter.KubernetesReporter {
+	// When using custom crds, we need to add them to the scheme
+	addToScheme := func(s *runtime.Scheme) error {
```
Comment (Tal-or) [inline test/e2e/k8sreporter/reporter.go:18]: What about machinconfig apis do we need them as well?

Comment (liornoy) [reply test/e2e/k8sreporter/reporter.go:18]: yes, we need everything that is related to this operator. I'll add. thanks!

**Comment by liornoy** [comment]
@Tal-or thanks for your comment.
I added the machine-config api scheme plus the MachineConfig and MachineConfigPool CRDS.
Also I moved the k8sreporter to a better location, under the utils dir

**Comment by mrniranjan** [comment]
@liornoy i was testing this by running ginkgo and it fails with below error:

 ```
ginkgo  --v -r -junit-report=performance.xml -requireSuite 1_performance/
 # github.com/openshift-kni/k8sreporter
../../../../../vendor/github.com/openshift-kni/k8sreporter/reporter.go:128:26: ambiguous selector r.clients.Nodes
```

**Comment by liornoy** [comment]
I think it was my bad that the commit with the bump was missing when I `unhold`ed this

**Comment by yanirq** [comment]
Thinking out loud here,   we apply the custom reporters here to **some** of our suits and not on **all** of them .
If the custom reporters add more value and clarity to test runs maybe we should add them to all the test suits (can be done in a different PR as well)

**Comment by liornoy** [comment]
@yanirq I just updated the description of this PR to include examples of the reports output,
I think that maybe we can merge this part, and after a while after you and other developers get
to work with these new custom reporters, you could decide if you are happy and want to add it
to the other suites, or maybe have comments for improvements.
Anyway, I'll be happy to help with this effort.

**Comment by ffromani** [comment]
/rename Functests: Add a custom junit and a k8s reporter

**Comment by liornoy** [comment]
Closing this one because we are thinking of a new direction, using the default junit reporter, but just processing its output differently.

**Pull request closed without merging**